_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,334 @@
+That being said, use at your own risk!
+
+![Cruise Control Frontend](/assets/images/posts/2019-08-08-cruise-control-frontend.png)
+
+
+## Motivation
+
+We have been getting a lot of requests for how to set up the Cruise Control UI with a Strimzi-managed Kafka cluster.
+Maybe you want a pretty dashboard to view your cluster status.
+Maybe you want to compare optimization proposals without having to open a text editor.
+Maybe you are just a visual person.
+Regardless of the reason, we want to provide some basic instructions for getting you started!
+
+## Prerequisites
+
+For starters we need a Strimzi-managed Kafka cluster with Cruise Control enabled.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-1.png)
+
+Since Strimzi does not support creating custom Cruise Control API user roles we need to disable HTTP basic authentication settings.
+
+We can do so by configuring Cruise Control within a `Kafka` resource like this:
+
+``` 
+# kafka-cruise-control.yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  cruiseControl:
+    config:
+      webserver.security.enable: false
+```
+
+After configuring our Strimzi-managed Kafka cluster, we can deploy and connect the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) in one of two ways:
+
+* METHOD ONE: Customizing the Cruise Control pod
+* METHOD TWO: Creating a dedicated Cruise Control UI pod
+
+## METHOD ONE: Customizing the Cruise Control pod
+
+With our Strimzi-managed Kafka cluster deployed, we can create and deploy a custom Cruise Control image which includes the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) application inside of it.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-1.png)
+
+
+### Inside the custom Cruise Control container
+
+Here the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) will run alongside the Cruise Control application in the same container making its requests directly to the Cruise Control REST API.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-3.png)
+
+### Building the Cruise Control UI container
+
+We can base our custom Cruise Control image on the original Strimzi Kafka image and install the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) bits on top:
+
+```bash
+#Dockerfile
+FROM quay.io/strimzi/kafka:latest-kafka-3.2.3
+
+ENV CC_UI_VERSION=0.4.0
+ENV CC_UI_HOME=./cruise-control-ui/dist/
+
+USER root
+
+RUN mkdir -p ${CC_UI_HOME}
+
+RUN curl -LO https://github.com/linkedin/cruise-control-ui/releases/download/v${CC_UI_VERSION}/cruise-control-ui-${CC_UI_VERSION}.tar.gz; \
+    tar xvfz cruise-control-ui-${CC_UI_VERSION}.tar.gz -C ${CC_UI_HOME} --strip-components=2; \
+    rm -f cruise-control-ui-${CC_UI_VERSION}.tar.gz*; \
+    echo "dev,dev,/kafkacruisecontrol/" > "${CC_UI_HOME}"static/config.csv;
+
+USER 1001
```

Comment:
Maybe I missed something ... but who and how starts or serves the UI in this case? You do not seem to modify the startup scripts.

Reply:
The diagram is good. But how does it find it? Does CC Server automatically watch for `CC_UI_HOME` env var? Or how do you tell it _there is a director here with the UI, serve it_? I'm interested myself, but It hink this might be interested for the users as well.

Reply:
It is the latter,  we unpack the Cruise Control UI files into the `/cruise-control-ui/dist/` directory.
This directory must match the value specified by the Cruise Control  `webserver.ui.diskpath` configuration, the directory Cruise Control checks for UI files to serve. The default value of Cruise Control's  `webserver.ui.diskpath` configuration is `/cruise-control-ui/dist/`. Does that make sense?

 Just added a new sentence to include this information

Reply:
Ok, makes sense. Thanks for the explanation. I think you should add `.` to the path => `./cruise-control-ui/dist/` as that is what you use in th eimage and what is the default in CC.